internal/controller/controllers/agentserviceconfig_controller.go

```diff
@@ -581,21 +608,23 @@ func monitorOperands(ctx context.Context, log logrus.FieldLogger, asc ASC) (stri
 		}
 	}
 
-	// monitor statefulset
-	ss := &appsv1.StatefulSet{}
-	if err := asc.Client.Get(ctx, types.NamespacedName{Name: imageServiceName, Namespace: asc.namespace}, ss); err != nil {
-		return "", err
-	}
+	// monitor statefulset (only if image service is not disabled)
+	if isImageServiceEnabled(asc.Object.GetAnnotations()) {
```

Comment:
If we split this into a separate function this can be a guard an will be a lot easier to read I think.